exercises/express-notes-api/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+
+const app = express();
+
+const fs = require('fs');
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Listening on port 3000!');
+});
+
+app.get('/api/notes', (req, res) => {
+  const notesData = [];
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).send({ error: 'an unexpected error occured' });
+      return;
+    }
+    const parsedData = JSON.parse(data);
+    for (const key in parsedData.notes) {
+      notesData.push(parsedData.notes[key]);
+    }
+    res.status(200).send(notesData);
+  });
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).send({ error: 'an unexpected error occured' });
+      return;
+    }
+    const parsedData = JSON.parse(data);
+    if (Math.sign(req.params.id) === 1 && Number.isInteger(parseFloat(req.params.id))) {
```

Comment:
It's better to do this check before you read the file so you can avoid using CPU/RAM for nothing.